### PR TITLE
ci: run codeql on pull requests

### PR DESCRIPTION
**Pull request by justinmk**
Try out codeql in PRs.

### CODE REVIEW

**Comment by dundargoc** [comment]
Don't we already use codeql?

**Comment by justinmk** [comment]
yeah we get security alerts from it. but I don't know how to check it against a PR (before merging the PR). is there a way?

**Comment by dundargoc** [comment]
Just add `pull_request` in the `on` field in file `.github/workflows/codeql.yml`.

**Comment by justinmk** [comment]
derp, I missed `.github/workflows/codeql.yml`. Thanks

**Comment by justinmk** [comment]
codeql job took 11 minutes, that doesn't seem too bad. Any objections to enabling it for PRs?

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -1,6 +1,11 @@
 name: "CodeQL"
 
 on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
```
Comment (dundargoc) [inline .github/workflows/codeql.yml:7]: Why? To my knowledge the push and pull_request events are completely separate/unrelated.

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -1,6 +1,11 @@
...
+    branches: [ "master" ]
   schedule:
```
Comment (dundargoc) [inline .github/workflows/codeql.yml:9]: Schedule shouldn't be needed anymore if we're testing this for every PR anyway.

**Comment by dundargoc** [comment]
No objections. I personally never found the warnings that useful which is why we never enabled it per PR IIRC. You could shave off a minute or two by caching the deps. Slapping on a

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (justinmk) [reply .github/workflows/codeql.yml:7]: idk, that's what the codeql default config had. should i remove it?

Comment (dundargoc) [reply .github/workflows/codeql.yml:7]: Up to you, I just find the comment confusing and misleading. The pull_request event will run on PR openings/updates and the push event will run when the PR is merged, and these are unrelated events. It should be possible to use only the push event or the pull_request event without the other.

Comment (dundargoc) [reply .github/workflows/codeql.yml:7]: Now that I think about it this doesn't really make much difference, this is an unnecessary nitpick. Disregard it. PR looks good to go.

**Pull request merged**